src/main/java/com/aws/greengrass/shadowmanager/configuration/ShadowDocSizeConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.shadowmanager.util.Validator;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC;
+import static com.aws.greengrass.shadowmanager.model.Constants.DEFAULT_DOCUMENT_SIZE;
+
+public final class ShadowDocSizeConfiguration {
+    @Getter
+    private final int maxShadowDocSizeConfiguration;
+
+    private ShadowDocSizeConfiguration(int maxShadowDocSizeConfiguration) {
+        this.maxShadowDocSizeConfiguration = maxShadowDocSizeConfiguration;
+    }
+
+    /**
+     * Creates a new shadow doc size configuration object and triggers updates based on previous configuration.
+     *
+     * @param serviceTopics    current configuration topics
+     * @return rate limits configuration objects
+     */
+    public static ShadowDocSizeConfiguration from(Topics serviceTopics) {
+        return new ShadowDocSizeConfiguration(getMaxShadowDocSizeFromConfig(serviceTopics));
+    }
+
+    private static int getMaxShadowDocSizeFromConfig(Topics topics) {
+        int newMaxShadowSize = Coerce.toInt(
+                topics.findOrDefault(DEFAULT_DOCUMENT_SIZE, CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC));
+        Validator.validateMaxShadowSize(newMaxShadowSize);
```

Comment:
Let's move the logic from the `Validator. validateMaxShadowSize` to a private method on this class. I explained in another comment why the validator class might not be well suited.